In [1]:
import numpy as np
import pandas  as pd
import statsmodels.api as sm
from sklearn import linear_model

1) Hava durumu modeli
Bu ödev için, geçmiş sıcaklık veri kümesine yeniden bakacaksınız. Aşağıdaki görevler için çözümünüzü içeren Jupyter notebooks'u gist dosyasına link olarak ekleyin.

*Öncelikle, Sıcaklık verilerini Kaggle'dan indirin.

*Önceki derste olduğu gibi, hedef değişkeninizin görünür sıcaklık ve sıcaklık arasındaki fark olduğu bir doğrusal regresyon modeli oluşturun. Açıklayıcı değişkenler olarak nem ve rüzgar hızı kullanın. Şimdi, modelinizi OLS kullanarak tahmin edin. R-kare ve ayarlanmış R-kare değerleri nelerdir? Tatminkar olduklarını düşünüyor musunuz? Açıklayın.

*Daha sonra, yukarıdaki modele nem ve rüzgar hızı etkileşimini dahil edin ve OLS'yi kullanarak modeli tahmin edin. Şimdi, bu modelin R-kare değeri nedir? Bu model bir öncekine göre gelişti mi?

*İlk modele ek açıklayıcı değişken olarak görünürlük ekleyin ve tahmin edin. R-kare arttı mı? Ayarlanmış R-kare değeri ne oldu? Tabloda ortaya çıkan farklılıkları, ayarlanmış R-kare içindeki iyileşme açısından etkileşim terimi ve görünürlük ile karşılaştırın. Hangisi daha kullanışlı?

*AIC ve BIC puanlarına göre yukarıdaki üç modelden en iyisini seçin. Mentor ile gerekçenizi tartışarak seçiminizi doğrulayın.

In [2]:
sicaklik = pd.read_csv('data/weatherHistory.csv')
sicaklik.columns

Index(['Formatted Date', 'Summary', 'Precip Type', 'Temperature (C)',
       'Apparent Temperature (C)', 'Humidity', 'Wind Speed (km/h)',
       'Wind Bearing (degrees)', 'Visibility (km)', 'Loud Cover',
       'Pressure (millibars)', 'Daily Summary'],
      dtype='object')

***Önceki derste olduğu gibi, hedef değişkeninizin görünür sıcaklık ve sıcaklık arasındaki fark olduğu bir doğrusal regresyon modeli oluşturun. Açıklayıcı değişkenler olarak nem ve rüzgar hızı kullanın. Şimdi, modelinizi OLS kullanarak tahmin edin. R-kare ve ayarlanmış R-kare değerleri nelerdir? Tatminkar olduklarını düşünüyor musunuz? Açıklayın.***

In [3]:
sicaklik_farki = sicaklik['Temperature (C)'] - sicaklik['Apparent Temperature (C)'] 

In [4]:
Y = sicaklik_farki
X = sicaklik[['Humidity','Wind Speed (km/h)']] 

In [5]:
lrm = linear_model.LinearRegression()
lrm.fit(X, Y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [6]:
print('Değişkenler: \n', lrm.coef_)
print('Sabit değer (bias): \n', lrm.intercept_)

Değişkenler: 
 [3.02918594 0.11929075]
Sabit değer (bias): 
 -2.4381054151877017


In [7]:
import statsmodels.api as sm

X = sm.add_constant(X)
results = sm.OLS(Y, X).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.288
Model:                            OLS   Adj. R-squared:                  0.288
Method:                 Least Squares   F-statistic:                 1.949e+04
Date:                Sat, 20 Apr 2019   Prob (F-statistic):               0.00
Time:                        21:37:39   Log-Likelihood:            -1.7046e+05
No. Observations:               96453   AIC:                         3.409e+05
Df Residuals:                   96450   BIC:                         3.409e+05
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -2.4381      0.021   -115.948      0.000      -2.479      -2.397
Humidity              3.0292      0.024    126.479      0.000       2.982       3.076
Wind Speed (km/h)     0.1193      0.001    176.164      0.000       0.118       0.121
==============================================================================
Omnibus:                     3935.747   Durbin-Watson:                   0.264
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4613.311
Skew:                           0.478   Prob(JB):                         0.00
Kurtosis:                       3.484   Cond. No.                         88.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

R kare ve ayarlanmis R kare degerlerimiz oldukca dusuk 0.288 bu yuzden modelimiz %28.8 lik bir oranla aciklandigini goruyoruz. %72.2 kadarlik kismi aciklanmamistir. 

***Daha sonra, yukarıdaki modele nem ve rüzgar hızı etkileşimini dahil edin ve OLS'yi kullanarak modeli tahmin edin. Şimdi, bu modelin R-kare değeri nedir? Bu model bir öncekine göre gelişti mi?***


In [8]:
sicaklik['Nem_Ruzgar_etkilesimi'] =sicaklik['Humidity'] * sicaklik['Wind Speed (km/h)']

In [9]:
Y = sicaklik_farki
X = sicaklik[['Humidity','Wind Speed (km/h)','Nem_Ruzgar_etkilesimi' ]] 

In [10]:
lrm = linear_model.LinearRegression()
lrm.fit(X, Y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [11]:
print('Değişkenler: \n', lrm.coef_)
print('Sabit değer (bias): \n', lrm.intercept_)

Değişkenler: 
 [-0.17751219 -0.09048213  0.29711946]
Sabit değer (bias): 
 -0.0839363100978221


In [12]:
import statsmodels.api as sm

X = sm.add_constant(X)
results = sm.OLS(Y, X).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.341
Model:                            OLS   Adj. R-squared:                  0.341
Method:                 Least Squares   F-statistic:                 1.666e+04
Date:                Sat, 20 Apr 2019   Prob (F-statistic):               0.00
Time:                        21:37:39   Log-Likelihood:            -1.6669e+05
No. Observations:               96453   AIC:                         3.334e+05
Df Residuals:                   96449   BIC:                         3.334e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=========================================================================================
                            coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------
const                    -0.0839      0.033     -2.511      0.012      -0.149      -0.018
Humidity                 -0.1775      0.043     -4.133      0.000      -0.262      -0.093
Wind Speed (km/h)        -0.0905      0.002    -36.797      0.000      -0.095      -0.086
Nem_Ruzgar_etkilesimi     0.2971      0.003     88.470      0.000       0.291       0.304
==============================================================================
Omnibus:                     4849.937   Durbin-Watson:                   0.262
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             9295.404
Skew:                           0.378   Prob(JB):                         0.00
Kurtosis:                       4.320   Cond. No.                         193.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Modele ekledigimiz yeni degisken ile R kare ve ayarlanmis R kare degerleri %34.1 oldu ancak halen dusuk. Bu az miktarda artisin aciklamayan bir varyansin yeni degiskenle aciklandigini gosterir. 

***İlk modele ek açıklayıcı değişken olarak görünürlük ekleyin ve tahmin edin. R-kare arttı mı? Ayarlanmış R-kare değeri ne oldu? Tabloda ortaya çıkan farklılıkları, ayarlanmış R-kare içindeki iyileşme açısından etkileşim terimi ve görünürlük ile karşılaştırın. Hangisi daha kullanışlı?***

In [13]:
Y = sicaklik_farki
X = sicaklik[['Humidity','Wind Speed (km/h)', 'Visibility (km)']] 

In [14]:
lrm = linear_model.LinearRegression()
lrm.fit(X, Y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [15]:
print('Değişkenler: \n', lrm.coef_)
print('Sabit değer (bias): \n', lrm.intercept_)

Değişkenler: 
 [ 2.60664109  0.11990113 -0.05398318]
Sabit değer (bias): 
 -1.5755946860023338


In [16]:
X = sm.add_constant(X)
results = sm.OLS(Y, X).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.304
Model:                            OLS   Adj. R-squared:                  0.303
Method:                 Least Squares   F-statistic:                 1.401e+04
Date:                Sat, 20 Apr 2019   Prob (F-statistic):               0.00
Time:                        21:37:39   Log-Likelihood:            -1.6938e+05
No. Observations:               96453   AIC:                         3.388e+05
Df Residuals:                   96449   BIC:                         3.388e+05
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -1.5756      0.028    -56.605      0.000      -1.630      -1.521
Humidity              2.6066      0.025    102.784      0.000       2.557       2.656
Wind Speed (km/h)     0.1199      0.001    179.014      0.000       0.119       0.121
Visibility (km)      -0.0540      0.001    -46.614      0.000      -0.056      -0.052
==============================================================================
Omnibus:                     3833.895   Durbin-Watson:                   0.279
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             4584.022
Skew:                           0.459   Prob(JB):                         0.00
Kurtosis:                       3.545   Cond. No.                         131.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Ilk modele Visibility (km) degiskenini eklememiz R kare ve ayarlanmis R kare yuzdesini dusurerek %30.4'e dusurdu. Boylece aciklanamayan varyanslar modelde az da olsa artti ve model daha anlamsiz hale geldi. 

***AIC ve BIC puanlarına göre yukarıdaki üç modelden en iyisini seçin. Mentor ile gerekçenizi tartışarak seçiminizi doğrulayın.***

Görünür sıcaklık ve sıcaklık arasındaki farkin bagimli degisken halindeki modellerde:

*1) Nem ve Ruzgar hizi degiskenlerimizle; AIC:	3.4090 ve BIC:3.4090*

*2) Nem ve rüzgar hızı etkileşimini dahil edilen modelde; AIC:3.3340 ve BIC:3.3340*

*3) Gorulen sicakligin eklendigi modelde; AIC:3.388 ve BIC:3.388*

Sonucta az da olsa yuksek R kare degeri yuzunden, AIC ve BIC degerleri neredeyse ayni olarak ilk modelimiz daha aciklayici duruyor. 


***2. Ev fiyatları modeli
Bu alıştırmada, ev fiyatları modelinizde çalışacaksınız. Aşağıdaki görevler için çözümünüzü içeren Jupyter notebooks'u gist dosyasına link olarak ekleyin.:***

*İlk olarak, Ev Fiyatları verilerini Kaggle'dan indirin.

*Ev fiyatları modelinizi tekrar çalıştırın ve F-testi, R-kare, ayarlanmış R-kare, AIC ve BIC kullanarak modelinizin uygunluğunu değerlendirin.

*Modelinizin tatmin edici olduğunu düşünüyor musunuz? Açıklayın

*Modelinizin uygunluğunu iyileştirmek için, bazı değişkenleri ekleyerek veya kaldırarak farklı model özelliklerini deneyin.

*Denediğiniz her model için, uygun metrikleri alın ve modellerinizi birbiriyle karşılaştırın. Hangi model en iyisidir ve neden?

In [17]:
ev_fiyatlari = pd.read_csv("data/train.csv")
ev_fiyatlari.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities    ...     PoolArea PoolQC Fence MiscFeature MiscVal  \
0         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   
1         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   
2         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   
3         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   
4         Lvl    AllPub    ...            0    NaN   NaN         NaN       0   

  MoSold YrSold  SaleType  SaleCondition  SalePrice  
0      2   2008        WD         Normal     208500  
1      5   2007        WD         Normal     181500  
2      9   2008        WD         Normal     223500  
3      2   2006        WD        Abnorml     140000  
4     12   2008        WD         Normal     250000  

[5 rows x 81 columns]

**Ev fiyatları modelinizi tekrar çalıştırın ve F-testi, R-kare, ayarlanmış R-kare, AIC ve BIC kullanarak modelinizin uygunluğunu değerlendirin.**

In [18]:
Y = ev_fiyatlari['SalePrice']

X = ev_fiyatlari[['YearBuilt', 'TotRmsAbvGrd', 'FullBath' , '1stFlrSF', 'TotalBsmtSF',
            'GarageArea', 'GarageCars', 'GrLivArea', 'OverallQual', 'YearRemodAdd']]

In [19]:
lrm = linear_model.LinearRegression()
lrm.fit(X, Y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [20]:
print('Değişkenler: \n', lrm.coef_)
print('Sabit değer (bias): \n', lrm.intercept_)

Değişkenler: 
 [ 2.68240707e+02  3.31050771e+01 -6.79087146e+03  1.41737355e+01
  1.98650991e+01  1.49475334e+01  1.04179010e+04  5.12971178e+01
  1.96045898e+04  2.96481161e+02]
Sabit değer (bias): 
 -1186193.7666661674


In [21]:
X = sm.add_constant(X)
results = sm.OLS(Y, X).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              SalePrice   R-squared:                       0.774
Model:                            OLS   Adj. R-squared:                  0.772
Method:                 Least Squares   F-statistic:                     495.4
Date:                Sat, 20 Apr 2019   Prob (F-statistic):               0.00
Time:                        21:37:39   Log-Likelihood:                -17459.
No. Observations:                1460   AIC:                         3.494e+04
Df Residuals:                    1449   BIC:                         3.500e+04
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const        -1.186e+06   1.29e+05     -9.187      0.000   -1.44e+06   -9.33e+05
YearBuilt      268.2407     50.346      5.328      0.000     169.481     367.000
TotRmsAbvGrd    33.1051   1119.061      0.030      0.976   -2162.048    2228.258
FullBath     -6790.8715   2682.369     -2.532      0.011   -1.21e+04   -1529.130
1stFlrSF        14.1737      4.930      2.875      0.004       4.504      23.844
TotalBsmtSF     19.8651      4.295      4.625      0.000      11.439      28.291
GarageArea      14.9475     10.311      1.450      0.147      -5.279      35.174
GarageCars    1.042e+04   3044.272      3.422      0.001    4446.249    1.64e+04
GrLivArea       51.2971      4.233     12.119      0.000      42.994      59.600
OverallQual    1.96e+04   1190.159     16.472      0.000    1.73e+04    2.19e+04
YearRemodAdd   296.4812     63.635      4.659      0.000     171.655     421.307
==============================================================================
Omnibus:                      477.814   Durbin-Watson:                   1.985
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            58906.279
Skew:                          -0.412   Prob(JB):                         0.00
Kurtosis:                      34.107   Cond. No.                     4.69e+05
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.69e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

**Modelinizin tatmin edici olduğunu düşünüyor musunuz? Açıklayın.**

*F-statistic:495.4 ve P degeri 0'a yakindir bu da modelin ev fiyatlarini aciklarken yararli oldugunu gosterir. 
AIC: 3.4940 ve BIC: 3.5000 Bu degerler daha dusuk hale yeni degiskenler ile getirilebilir.*

**Modelinizin uygunluğunu iyileştirmek için, bazı değişkenleri ekleyerek veya kaldırarak farklı model özelliklerini deneyin.**

In [22]:
corr_matrix = ev_fiyatlari.corr()
corr_matrix.SalePrice.sort_values(ascending=False)[1:20]

OverallQual     0.790982
GrLivArea       0.708624
GarageCars      0.640409
GarageArea      0.623431
TotalBsmtSF     0.613581
1stFlrSF        0.605852
FullBath        0.560664
TotRmsAbvGrd    0.533723
YearBuilt       0.522897
YearRemodAdd    0.507101
GarageYrBlt     0.486362
MasVnrArea      0.477493
Fireplaces      0.466929
BsmtFinSF1      0.386420
LotFrontage     0.351799
WoodDeckSF      0.324413
2ndFlrSF        0.319334
OpenPorchSF     0.315856
HalfBath        0.284108
Name: SalePrice, dtype: float64

In [23]:
Y = ev_fiyatlari['SalePrice']

X = ev_fiyatlari[['YearBuilt', 'TotRmsAbvGrd', 'FullBath' , '1stFlrSF', 'TotalBsmtSF',
                  'GarageArea', 'GarageCars', 'GrLivArea', 'OverallQual', 'YearRemodAdd', 'MasVnrArea', 'Fireplaces',
                  'BsmtFinSF1','LotFrontage', 'WoodDeckSF','2ndFlrSF', 'OpenPorchSF', 'HalfBath']]

In [24]:
X.isnull().any()

YearBuilt       False
TotRmsAbvGrd    False
FullBath        False
1stFlrSF        False
TotalBsmtSF     False
GarageArea      False
GarageCars      False
GrLivArea       False
OverallQual     False
YearRemodAdd    False
MasVnrArea       True
Fireplaces      False
BsmtFinSF1      False
LotFrontage      True
WoodDeckSF      False
2ndFlrSF        False
OpenPorchSF     False
HalfBath        False
dtype: bool

Bu 3 degiskende NaN degerler oldugu icin hangi turde olduklarina bakip: 

*1) GarageYrBlt, float64# 
   MasVnrArea, float64# 
   LotFrontage, float64#* 

*2)GarageYrBlt sutununu mean ile*

*3) Diger iki degiskeni de mode ile doldurabiliriz.*

**Modele GarageYrBlt sutunu dahil edilmedi.**

In [25]:
fill_list = ['MasVnrArea', 'LotFrontage']

In [26]:
for col in fill_list:
    X.loc[:, col].fillna(X.loc[:, col].mean(), inplace=True)

C:\Users\ece-yusuf-sekoya\Anaconda3\lib\site-packages\pandas\core\generic.py:5434: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)


In [27]:
ev_fiyatlari.GarageYrBlt.fillna(ev_fiyatlari.GarageYrBlt.mode()[0])

0       2003.0
1       1976.0
2       2001.0
3       1998.0
4       2000.0
5       1993.0
6       2004.0
7       1973.0
8       1931.0
9       1939.0
10      1965.0
11      2005.0
12      1962.0
13      2006.0
14      1960.0
15      1991.0
16      1970.0
17      1967.0
18      2004.0
19      1958.0
20      2005.0
21      1930.0
22      2002.0
23      1976.0
24      1968.0
25      2007.0
26      2005.0
27      2008.0
28      1957.0
29      1920.0
         ...  
1430    2005.0
1431    1976.0
1432    1928.0
1433    2000.0
1434    1977.0
1435    1962.0
1436    1974.0
1437    2008.0
1438    1957.0
1439    1979.0
1440    1993.0
1441    2004.0
1442    2008.0
1443    1916.0
1444    2004.0
1445    1990.0
1446    1962.0
1447    1995.0
1448    1950.0
1449    2005.0
1450    2005.0
1451    2008.0
1452    2005.0
1453    2005.0
1454    2004.0
1455    1999.0
1456    1978.0
1457    1941.0
1458    1950.0
1459    1965.0
Name: GarageYrBlt, Length: 1460, dtype: float64

In [28]:
X.isnull().any()

YearBuilt       False
TotRmsAbvGrd    False
FullBath        False
1stFlrSF        False
TotalBsmtSF     False
GarageArea      False
GarageCars      False
GrLivArea       False
OverallQual     False
YearRemodAdd    False
MasVnrArea      False
Fireplaces      False
BsmtFinSF1      False
LotFrontage     False
WoodDeckSF      False
2ndFlrSF        False
OpenPorchSF     False
HalfBath        False
dtype: bool

In [29]:
lrm = linear_model.LinearRegression()
lrm.fit(X, Y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [30]:
print('Değişkenler: \n', lrm.coef_)
print('Sabit değer (bias): \n', lrm.intercept_)

Değişkenler: 
 [ 1.69603304e+02  1.60164397e+03 -2.22087629e+03  2.58522709e+01
  1.12283383e+01  9.14211074e+00  1.03551741e+04  2.06155841e+01
  1.89074239e+04  3.55432531e+02  2.87239893e+01  7.65755267e+03
  1.72710024e+01  5.67519239e+01  3.23079353e+01  1.89497244e+01
  1.08092192e+01 -1.47521367e+03]
Sabit değer (bias): 
 -1105848.5143657622


In [31]:
X = sm.add_constant(X)
results = sm.OLS(Y, X).fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:              SalePrice   R-squared:                       0.792
Model:                            OLS   Adj. R-squared:                  0.789
Method:                 Least Squares   F-statistic:                     304.8
Date:                Sat, 20 Apr 2019   Prob (F-statistic):               0.00
Time:                        21:37:40   Log-Likelihood:                -17398.
No. Observations:                1460   AIC:                         3.483e+04
Df Residuals:                    1441   BIC:                         3.493e+04
Df Model:                          18                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const        -1.106e+06   1.32e+05     -8.402      0.000   -1.36e+06   -8.48e+05
YearBuilt      169.6033     53.314      3.181      0.001      65.021     274.186
TotRmsAbvGrd  1601.6440   1094.633      1.463      0.144    -545.600    3748.888
FullBath     -2220.8763   2852.204     -0.779      0.436   -7815.793    3374.040
1stFlrSF        25.8523     20.859      1.239      0.215     -15.065      66.770
TotalBsmtSF     11.2283      4.309      2.606      0.009       2.775      19.682
GarageArea       9.1421     10.150      0.901      0.368     -10.767      29.052
GarageCars    1.036e+04   2974.862      3.481      0.001    4519.651    1.62e+04
GrLivArea       20.6156     20.404      1.010      0.312     -19.409      60.640
OverallQual   1.891e+04   1173.946     16.106      0.000    1.66e+04    2.12e+04
YearRemodAdd   355.4325     62.210      5.713      0.000     233.400     477.465
MasVnrArea      28.7240      6.172      4.654      0.000      16.617      40.831
Fireplaces    7657.5527   1787.219      4.285      0.000    4151.722    1.12e+04
BsmtFinSF1      17.2710      2.594      6.658      0.000      12.182      22.360
LotFrontage     56.7519     49.526      1.146      0.252     -40.398     153.902
WoodDeckSF      32.3079      8.136      3.971      0.000      16.348      48.268
2ndFlrSF        18.9497     20.658      0.917      0.359     -21.573      59.472
OpenPorchSF     10.8092     15.739      0.687      0.492     -20.064      41.683
HalfBath     -1475.2137   2774.803     -0.532      0.595   -6918.299    3967.871
==============================================================================
Omnibus:                      654.551   Durbin-Watson:                   1.970
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           105699.602
Skew:                          -0.991   Prob(JB):                         0.00
Kurtosis:                      44.636   Cond. No.                     5.05e+05
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 5.05e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

**Denediğiniz her model için, uygun metrikleri alın ve modellerinizi birbiriyle karşılaştırın. Hangi model en iyisidir ve neden?**

*10 aciklayici degiskenli bir onceki modelde -->  F-statistic:495.4 ve P degeri 0'a yakindir bu da modelin ev fiyatlarini aciklarken yararli oldugunu gosterir. 
*AIC: 3.4940 ve BIC: 3.5000 Bu degerler daha dusuk hale yeni degiskenler ile getirilebilir.R kare degeri ise %77 dir*

*Bu 18 degiskenli modelimizde --> F-statistic:304.8 ve P degeri 0'a yakindir bu da modelin ev fiyatlarini aciklarken yararli oldugunu gosterir. 
*AIC:3.4830  ve BIC: 3.4930 olarak ilk modele gore biraz daha azalmistir. R kare degeri ise %79 olarak eklenen degiskenlerin modelin ev fiyatlarini aciklamasinda 2 birimlik bir artis sagladigini goruyoruz.*